utils/lib/Component.ts

```diff
@@ -0,0 +1,51 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+
+abstract class Component<ConfigType, MutationVariablesType> {
+  public localPath: string; // Local path to the component. Ex: python/flask
```

Comment:
I worry that this variable isn't super clear what it represents. Perhaps we should call this `identifier`? `slug`?

Reply:
Updated to use `identifier`